src/components/transmuters/EthTransmuterLooper.tsx

```diff
@@ -0,0 +1,652 @@
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "lastReport",
+        chainId: chain.id,
+      },
+      {
+        // Get the current balance in y shares of the address or account.
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "balanceOf",
+        chainId: chain.id,
+        args: [address!],
+      },
+      {
+        // Get the price in asset per share
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "pricePerShare",
+        chainId: chain.id,
+      },
+      {
+        // Gets the current time profits are set to unlock over as uint.
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "profitMaxUnlockTime",
+        chainId: chain.id,
+      },
+      {
+        // Gets the current time profits are set to unlock over as uint.
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "profitUnlockingRate",
+        chainId: chain.id,
+      },
+      {
+        // Gets the total assets deposited
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "totalAssets",
+        chainId: chain.id,
+      },
+      {
+        // Gets the total shares outstanding
+        address: TRANSMUTER_LOOPER_ADDRESS,
+        abi: tokenizedStrategyAbi,
+        functionName: "totalSupply",
+        chainId: chain.id,
+      },
+    ],
+    query: {
+      select: ([
+        fullProfitUnlockDate,
+        isShutdown,
+        lastReport,
+        balanceOf,
+        pricePerShare,
+        profitMaxUnlockTime,
+        profitUnlockingRate,
+        totalAssets,
+        totalSupply,
+      ]) => {
+        return [
+          BigInt(fullProfitUnlockDate as number),
+          isShutdown,
+          BigInt(lastReport as number),
+          BigInt(balanceOf as number),
+          BigInt(pricePerShare as number),
+          BigInt(profitMaxUnlockTime as number),
+          BigInt(profitUnlockingRate as number),
+          BigInt(totalAssets as number),
+          BigInt(totalSupply as number),
+        ];
+      },
+    },
+  });
+
+  // Watch intermittently updated read values in case they are updated due to the report function or other manual updates to the contract
+  useWatchQuery({
+    scopeKey: ScopeKeys.TransmuterLooperReported,
+  });
+
+  /** CONTRACT WRITES **/
+  /**
+   * Get allowance info for alETH
+   */
+  const {
+    isApprovalNeeded,
+    approveConfig: alETHApproveConfig,
+    approve: approveSpendAlETH,
+  } = useAllowance({
+    tokenAddress: SYNTH_ASSETS_ADDRESSES[mainnet.id][SYNTH_ASSETS.ALETH],
+    spender: TRANSMUTER_LOOPER_ADDRESS,
+    amount,
+    decimals:
+      (transmuterLooperContractStaticState?.[
+        TL_STATIC_STATE_INDICIES.decimals
+      ] as number) || 18,
```

Comment:
when you will have destructed the `transmuterLooperContractStaticState` you would simply put `decimals` here. Not `undefined` state is handled in `useAllowance`.